### ISSUE TITLE
Mouse scrolling is not detected after closing a window until the mouse is moved

### ISSUE THREAD

**Issue by vE5li** [hyprwm/Hyprland#3679]
### Hyprland Version

0.30.0

### Bug or Regression?

Bug

### Description

Very similar to this issue https://github.com/hyprwm/Hyprland/issues/1675

### Steps to reproduce:
- Open two windows
- Put your mouse over one of them and close it (cursor should be on the other window now)
- Scroll without moving the mouse

### Expected outcome:
The window should scroll

### Noted outcome:
It doesn't. However, the scrolling continues after I move my mouse inside the window.

Since I mostly scroll using my keyboard this is happening quite a lot for me. I would also be happy to create a PR myself depending on what the feedback on this issue is

**Comment by vaxerski** [comment]
This was fixed in -git.

**Comment by tchofy** [comment]
I've been able to replicate this on both OP's situation, and also with grouped windows, but only happens if `folow_mouse` is set to anything other than 0. Built from latest git as of writing (7f4b0aa).

**Comment by vaxerski** [comment]
my bad

**Comment by tchofy** [comment]
From some testing with wev and groups, seems like scroll events are being sent to the previous window, instead of the newly focused window. The behavior might be the same with closed windows.
https://github.com/hyprwm/Hyprland/assets/64620317/48e8559a-ab20-4bd0-abb1-a7195c2429ae

**Comment by vaxerski** [comment]
among us indeed. I know whats up

**Comment by vE5li** [comment]
I just realized the same thing is happening when opening a window, the previous window gets the scroll events

**Comment by tchofy** [comment]
I can still replicate this bug with some grouped windows, but seems like it's only for native wayland, xwayland works as expected. I'll do some more testing and open a new issue if I can confirm this.

**Comment by vE5li** [comment]
This is still an issue. I am unable to determine what the logic is exactly but something very weird is happening that is easy to reproduce:
- Open a single window on the workspace
- Scroll without moving the mouse

The scroll input will only be propagated correctly if my mouse cursor is the bottom right quarter of the window. It's extremely weird but I have tested extensively and at least on my device this is the case 100% of the time.

This behavior is consistent also when opening additional windows, every time my mouse cursor is on the bottom right it will get the scroll input correctly, otherwise it will not.

### Closing windows
I also noticed that closing a window still sometimes behaves as it did before a4db48b46b9b92b5a04b1f34490cb38802942656. Just to reiterate, open two windows, close one of them (cursor should be over the other one now now), try to scroll without moving the mouse.

For some windows in this scenario, like my terminal emulator (`foot`), the scroll event is never passed on properly, while for windows like `firefox` it works as expected. Both of them are native Wayland and I tried looking at the output of `hyprctl clients` to find some difference between them, but I could not.

I don't really understand what's going on here but the described behavior makes Hyprland feel a bit inconsistent and frustrating.

**Comment by vE5li** [comment]
@vaxerski Is it possible to reopen this? The issue is still very much present

**Comment by soufhehe** [comment]
it works, can be closed

**Comment by vE5li** [comment]
While the setup I described originally works now, altering the steps slightly still produces incorrect results. Sticking with the way I wrote the original ticket it would be:

- Open three windows (e.g. two left, one right)
- Focus one of the left windows
- Put your mouse over the right window
- Close the focused window
- Scroll without moving the mouse

As in the original setup, no scrolling event is received by the hovered window.

Should I open a new issue, given it's a different setup?
